src/Component/src/Symfony/Security/OperationAccessChecker.php

```diff
@@ -0,0 +1,89 @@
+<?php
+
+/*
+ * This file is part of the Sylius package.
+ *
+ * (c) Sylius Sp. z o.o.
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace Sylius\Resource\Symfony\Security;
+
+use Sylius\Resource\Context\Context;
+use Sylius\Resource\Metadata\Operation;
+use Sylius\Resource\Metadata\OperationAccessCheckerInterface;
+use Symfony\Component\ExpressionLanguage\ExpressionLanguage;
+use Symfony\Component\Security\Core\Authentication\AuthenticationTrustResolverInterface;
+use Symfony\Component\Security\Core\Authentication\Token\NullToken;
+use Symfony\Component\Security\Core\Authentication\Token\Storage\TokenStorageInterface;
+use Symfony\Component\Security\Core\Authentication\Token\TokenInterface;
+use Symfony\Component\Security\Core\Authorization\AuthorizationCheckerInterface;
+use Symfony\Component\Security\Core\Role\RoleHierarchyInterface;
+
+final class OperationAccessChecker implements OperationAccessCheckerInterface
+{
+    public function __construct(
+        private readonly ?ExpressionLanguage $expressionLanguage = null,
+        private readonly ?AuthenticationTrustResolverInterface $authenticationTrustResolver = null,
+        private readonly ?RoleHierarchyInterface $roleHierarchy = null,
+        private readonly ?TokenStorageInterface $tokenStorage = null,
+        private readonly ?AuthorizationCheckerInterface $authorizationChecker = null,
+    ) {
+    }
+
+    public function isGranted(Operation $operation, Context $context, array $extraVariables = []): bool
+    {
+        if (null === $this->tokenStorage || null === $this->authenticationTrustResolver) {
+            throw new \LogicException('The "symfony/security" library must be installed to use the "security" attribute.');
+        }
+
+        if (null === $this->expressionLanguage) {
+            throw new \LogicException('The "symfony/expression-language" library must be installed to use the "security" attribute.');
+        }
+
+        if (null === $expression = $operation->getSecurity()) {
+            return true;
+        }
```

Comment:
While I personally enjoy this way of getting and checking value at the same time -- it's considered a kind of anti-pattern lately and should be avoided used in public libraries.

```suggestion
        $expression = $operation->getSecurity();
        if (null === $expression) {
            return true;
        }
```

![image](https://github.com/user-attachments/assets/a0e14b8c-3544-4b85-848e-e10481be8198)